### PR TITLE
Consensus Testing Refactor: moving wallet / signature functions to ConsensusContext

### PR DESCRIPTION
**Pull request by igormcoelho**
To allow a viable and strong unit testing on ConsensusService (#436), we need to delegate some of its current tasks. I believe wallet/signing decisions would fit better on ConsensusContext class.

### CODE REVIEW

**Comment by igormcoelho** [comment]
@erikzhang @shargon @vncoelho this change simplifies ConsensusService by moving signing functions to ConsensusContext. Now, wallet and keypair are private variables, which is also nice. This allows us to test ConsensusService class without worrying about any signature process.

**Comment by erikzhang** [comment]
@igormcoelho I made some change. Can you look at this? Is it ok?

File: neo/Consensus/ConsensusContext.cs [hunk 1]
```diff
@@ -56,6 +56,23 @@ public void ChangeView(byte view_number)
...
+                if (Signatures[i] != null)
+                {
+                    sc.AddSignature(contract, Validators[i], Signatures[i]);
+                    j++;
+                }
+            sc.Verifiable.Witnesses = sc.GetWitnesses();
+            block.Transactions = TransactionHashes.Select(p => Transactions[p]).ToArray();
+            return block;
+        }
+
```
Comment (igormcoelho) [inline neo/Consensus/ConsensusContext.cs]: Very good! Simple and useful.

File: neo/Consensus/ConsensusService.cs [hunk 2]
```diff
@@ -83,8 +83,7 @@ private void CheckSignatures()
         {
             if (context.Signatures.Count(p => p != null) >= context.M && context.TransactionHashes.All(p => context.Transactions.ContainsKey(p)))
             {
-                Block block = context.MakeSignedHeader();
-                block.Transactions = context.TransactionHashes.Select(p => context.Transactions[p]).ToArray();
+                Block block = context.CreateBlock();
```
Comment (igormcoelho) [inline neo/Consensus/ConsensusService.cs]: Single line now.

File: neo/Consensus/ConsensusContext.cs [hunk 3]
```diff
@@ -138,6 +129,11 @@ private ConsensusPayload MakeSignedPayload(ConsensusMessage message)
             return payload;
         }
 
+        public void SignHeader()
+        {
+            Signatures[MyIndex] = MakeHeader()?.Sign(KeyPair);
+        }
+
```
Comment (igormcoelho) [inline neo/Consensus/ConsensusContext.cs]: Single line code, very good.

File: neo/Consensus/ConsensusContext.cs [hunk 4]
```diff
@@ -231,6 +231,7 @@ public void Fill()
             TransactionHashes = transactions.Select(p => p.Hash).ToArray();
             Transactions = transactions.ToDictionary(p => p.Hash);
             NextConsensus = Blockchain.GetConsensusAddress(Snapshot.GetValidators(transactions).ToArray());
+            Timestamp = Math.Max(DateTime.UtcNow.ToTimestamp(), Snapshot.GetHeader(PrevHash).Timestamp + 1);
```
Comment (igormcoelho) [inline neo/Consensus/ConsensusContext.cs]: Nice change!

**Comment by igormcoelho** [comment]
Very good Erik! I liked the CreateBlock function, now it's simple and concise.
@vncoelho @shargon please take a look at these small changes... ConsensusService is becoming smaller, more solid, and giving more auxiliary tasks to ConsensusContext.

**Pull request merged**